src/components/api.js

```diff
@@ -0,0 +1,146 @@
+export class Api {
+	/**
+	 * param: options
+	 * Description: constructor takes a options obj that carries a bae url
+	 *                      and headers that contain auth and content-type
+	 */
+	constructor(options) {
+		this._baseUrl = options.baseUrl;
+		this._header = options.headers;
+	} //end constructor
+
+	// Helper function to handle fetch response and errors
+	_handleResponse(res) {
+		if (res.ok) {
+			return res.json();
+		} //end if
+		return Promise.reject(`Error: ${res.status}`);
+	} //end func
+
+	/****************Card routes***************/
+	// 1. GET /cards – Get all cards
+	/**
+	 * Description: gets all initial cards form url
+	 *                      json obj returned if status 200
+	 */
+	getInitialCards() {
+		return fetch(`${this._baseUrl}/cards`, {
+			method: "GET",
+			headers: this._header,
+		})
+			.then(this._handleResponse)
+			.catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`    after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.